apostrophecms/apostrophe
# Add "busy" and "saving..saved" indicators

I added an overlay for HTTP requests when `options.busy` is passed.

I used an arbitrary color (to be checked with @stuartromanek later).

## How to test

Check out this branch, and run `npm link`.
Clone `a3-test-1` project , switch to `1940-new-module-format` branch, run `npm i` if not done previously, then run `npm link apostrophe`.

Go to a Vue component using a `http` request. For example, `lib/modules/@apostrophecms/areas/src/apos/components/ApostropheAreaEditor.vue` and modify the `up` method this way:

```js
   await apos.http.patch(`${apos.docs.action}/${this._docId}`, {
        busy: true,
        body: {
          $move: {
            [`@${this._id}.items`]: {
              $item: this.next[i].widget._id,
              $before: this.next[i - 1].widget._id
            }
          }
        }
      });
```

Add two widgets in the same area, and click on the "Up" button. As the overlay will be quickly removed and therefore hard to see, you can comment this part in `lib/modules/@apostrophecms/utils/src/apos/always/http.js` for test purposes and see the overlay is active (line 201):

```js
    xmlhttp.addEventListener('loadend', function () {
      if (options.busy) {
        apos.bus.$emit('apos-busy', false);
      }
    });
```
